### PR TITLE
8252721: Nested classes in Swing APIs rely on default constructors

### PR DESCRIPTION
**Pull request by prsadhuk**
Please review a fix for issue where it was seen that several 
nested classes rely on default constructors as part of their public API.

It's to be noted that "A no-arg public constructor is generated by the compiler for a class if it does not declare an explicit constructor. While convenient, this is inappropriate for many kinds of formal classes, both because the constructor will have no javadoc and because the constructor may be unintended."

For the JDK, classes intended to be used outside of the JDK, public classes in exported packages, should not rely on default constructors.

Proposed fix is to add explicit public no-arg constructors for public classes and protected no-arg constructor for public abstract classes for javax.swing module (as one part of overalll java.desktop change) 

CSR: https://bugs.openjdk.java.net/browse/JDK-8252908

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8252721](https://bugs.openjdk.java.net/browse/JDK-8252721): Nested classes in Swing APIs rely on default constructors

### Reviewers
 * [Phil Race](https://openjdk.java.net/census#prr) (@prrace - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/72/head:pull/72`
`$ git checkout pull/72`

### CODE REVIEW

File: src/java.desktop/share/classes/javax/swing/Box.java [hunk 1]
```diff
@@ -425,6 +430,11 @@ public AccessibleContext getAccessibleContext() {
      */
     @SuppressWarnings("serial")
     protected class AccessibleBox extends AccessibleAWTContainer {
+        /**
+         * Constructs a {@code AccessibleBox}.
+         */
+        public AccessibleBox() {}
```
Comment (mrserb) [inline src/java.desktop/share/classes/javax/swing/Box.java:436]: Shouldn't this be a protected constructor?
"if the class is declared protected, then the default constructor is implicitly given the access modifier protected".

Comment (prsadhuk) [reply src/java.desktop/share/classes/javax/swing/Box.java]: Modified to "protected" constructor

**Review by prrace — APPROVED** [review]

File: src/java.desktop/share/classes/javax/swing/Box.java (hunk 1 above)
Comment (mrserb) [reply src/java.desktop/share/classes/javax/swing/Box.java]: These classes are not abstract so not sure about the spec for the constructor: "Constructor for subclasses to call." Some of them like "AccessibleBox" is instantiated directly.

Comment (prsadhuk) [reply src/java.desktop/share/classes/javax/swing/Box.java]: Let's see what Joe says about the CSR spec...I will change if needed.

**Pull request closed without merging**